Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: Goal is to not require this function at all.
+    command = CommandsCommand.new
+    command.check_for_errors
```

Comment:
@woodruffw, if `check_for_errors` is put inside initialize, then `cmd/commands_spec.rb` fails. If it is inside `CommandsCommand#initialize()` function, then when `commands_spec.rb` tests individual methods such as in line 43, `cmds = described_class.new.internal_commands`, where `ARGV` is not set nor provided, it outputs this error message and terminates the script:
```
Error: 5 invalid options provided: --color --require --format --out --tty

Correct usage:
brew commands [--quiet [--include-aliases]]:
    Show a list of built-in and external commands.

    If --quiet is passed, list only the names of commands without the header.
    With --include-aliases, the aliases of internal commands will be included.

```

Thats why `check_for_errors ` needs to be out of the initialization.